src/main/java/backend/Logic.java

```diff
@@ -200,18 +198,51 @@ public Model getRepo(String repoId) {
      * @return true if label replacement on GitHub was a success, false otherwise.
      */
     public CompletableFuture<Boolean> replaceIssueLabels(TurboIssue issue, List<String> newLabels) {
+        List<String> originalLabels = issue.getLabels();
+
         logger.info("Changing labels for " + issue + " on UI");
-        issue.setLabels(newLabels);
+        Optional<TurboIssue> replaceResult = models.replaceIssueLabels(issue.getRepoId(), issue.getId(), newLabels);
+        if (!replaceResult.isPresent()) {
+            return CompletableFuture.completedFuture(false);
+        }
```

Comment:
These 4 lines looks like a high-level step. Can extract or at least add a comment to state what the step is?